### PR TITLE
Added table of content to source-generators.cookbook.md

### PR DESCRIPTION
**Pull request by AnderssonPeter**
Added table of content

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/roslyn?pullRequest=66317) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/roslyn?pullRequest=66317) before we can accept your contribution.<br/><br/>:x: AnderssonPeter [sign now](https://cla.dotnetfoundation.org/dotnet/roslyn?pullRequest=66317)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/roslyn?pullRequest=66317) it.</sub>

File: docs/features/source-generators.cookbook.md [hunk 1]
```diff
@@ -11,6 +11,33 @@ of scope in the final design of the shipping feature.
 
 **This document expands on the details in the [full design document](source-generators.md), please ensure you have read that first.**
 
+## Table of content
```
Comment (Youssef1313) [inline docs/features/source-generators.cookbook.md:14]:
```suggestion
## Table of content

```

**Review by Youssef1313 — APPROVED** [review]
LGTM

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 2)

**Pull request merged**